packages/components/src/mobile/link-settings/test/edit.native.js

```diff
@@ -0,0 +1,432 @@
+import { fireEvent, initializeEditor, waitFor } from 'test/helpers';
+/**
+ * WordPress dependencies
+ */
+import { registerCoreBlocks } from '@wordpress/block-library';
+import { getBlockTypes, unregisterBlockType } from '@wordpress/blocks';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Utility function to unregister all core block types previously registered
+ * when staging the Redux Store `beforeAll` integration tests start running.
+ *
+ * This should probably be extracted into a utility module for future tests.
+ */
+const unregisterBlocks = () => {
+	const blocks = getBlockTypes();
+
+	blocks.forEach( ( { name } ) => unregisterBlockType( name ) );
+};
+
+/**
+ * ### TODO
+ * + Try to figure out why I can't `console.log(JSON.stringify(subject.toJSON()))` anymore.
+ */
+describe.each( [
+	[
+		{
+			type: 'core/button',
+			initialHtml: `
+				<!-- wp:button {"style":{"border":{"radius":"5px"}}} -->
+				<div class="wp-block-button">
+					<a class="wp-block-button__link" style="border-radius:5px">Link</a>
+				</div>
+				<!-- /wp:button -->
+			`,
+			toJSON: () => 'core/button',
+		},
+	],
+	[
+		{
+			type: 'core/image',
+			initialHtml: `
+				<!-- wp:image {"id":20,"sizeSlug":"large","linkDestination":"custom"} -->
+				<figure class="wp-block-image size-large">
+					<img class="wp-image-20" src="https://tonytahmouchtest.files.wordpress.com/2021/10/img_0111-2.jpg?w=1024" alt="" />
+				</figure>
+				<!-- /wp:image -->
+			`,
+			toJSON: () => 'core/image',
+		},
+	],
+] )( '<LinkSettings/> from %j', ( { type, initialHtml } ) => {
+	beforeAll( () => {
+		registerCoreBlocks();
+	} );
+
+	afterAll( () => {
+		unregisterBlocks();
+	} );
+
+	/**
+	 * GIVEN an EDITOR is displayed with an EDIT IMAGE BLOCK or EDIT BUTTON BLOCK;
+	 * GIVEN the CLIPBOARD has a URL copied;
+	 * WHEN the USER selects the SETTINGS BUTTON on the EDIT IMAGE BLOCK or EDIT BUTTON BLOCK;
+	 */
+	// eslint-disable-next-line jest/no-done-callback
+	it( 'should display the LINK SETTINGS with an EMPTY LINK TO field.', async ( done ) => {
+		// Arrange
+		const expectation =
+			'The LINK SETTINGS > LINK TO field SHOULD be displayed WITHOUT a URL from the CLIPBOARD.';
+		const url = 'https://tonytahmouchtest.files.wordpress.com';
+		const subject = await initializeEditor( { initialHtml } );
+		Clipboard.getString.mockReturnValue( url );
+
+		// Act
+		try {
+			const block = await waitFor( () =>
+				subject.getByA11yLabel(
+					type === 'core/image' ? /Image Block/ : /Button Block/
+				)
+			);
+			fireEvent.press( block );
+			fireEvent.press( block );
+			fireEvent.press(
+				await waitFor( () => subject.getByA11yLabel( 'Open Settings' ) )
+			);
+		} catch ( error ) {
+			done.fail( error );
+		}
+
+		// Assert
+		try {
+			await waitFor( () =>
+				subject.getByA11yLabel(
+					`Link to, ${
+						type === 'core/image' ? 'None' : 'Search or type URL'
+					}`
+				)
+			);
+			done();
+		} catch ( error ) {
+			done.fail( expectation );
+		}
```

Comment:
@ttahmouch I'm out of curiosity about the use of `try-catch` in these tests, in case we could replicate it on future tests. I understand that the purpose is to control the message displayed when the test fails, right?

Reply:
That's correct. I didn't feel that the default error message is particularly expressive or helpful for the final test expectation. Though I considered just decorating the original `{Error}` instance emitted by `react-native-testing-library`. I can still consider changing it to log the original error as well if you think it would be helpful.

WDYT?

Reply:
No, I think it's great providing more accurate error messages, although it shouldn't be the common case as tests should succeed 😅 . Checking the Action section, I'm wondering if it's necessary the first `try-catch` there because if any error is raised the test would fail in any case before entering the Assert section, right?.